File: next-main/src/main/java/com/next/reservations/core/repository/ReservationTimeConfigurationRepository.java
package com.next.reservations.core.repository;

import com.next.reservations.core.domain.ReservationTimeConfiguration;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ReservationTimeConfigurationRepository extends JpaRepository<ReservationTimeConfiguration, Long> {
    ReservationTimeConfiguration findByDefaultConfigTrue();
    Optional<ReservationTimeConfiguration> findByDefaultStartDateIsNotNull();
}

File: next-main/src/main/kotlin/com/next/reservations/core/repository/ReservationTimeConfigurationRepository.kt
package com.next.reservations.core.repository

import com.next.reservations.core.domain.ReservationTimeConfiguration
import org.springframework.data.jpa.repository.JpaRepository

interface ReservationTimeConfigurationRepository : JpaRepository<ReservationTimeConfiguration, Long> {
    fun findByDefaultConfigTrue(): ReservationTimeConfiguration
    fun findByDefaultStartDateIsNotNull(): ReservationTimeConfiguration?
}
